Generals/Code/GameEngine/Source/GameLogic/Object/Update/AIUpdate/ChinookAIUpdate.cpp

```diff
@@ -587,6 +590,8 @@ class ChinookCombatDropState : public State
 				Object* rappeller = getPotentialRappeller(obj);
 				if (rappeller != NULL)
 				{
+					rappeller->setStatus(MAKE_OBJECT_STATUS_MASK(OBJECT_STATUS_IS_USING_ABILITY));
```

Comment:
Should this be added AFTER confirming door is available? 
If I understand correctly, when a door is not available, the status will never get cleared (the loop only clears units in the tracking list), so the unit could get stuck with the flag.

Reply:
The door cannot be reserved unless `getAiFreeToExit` returns true. If you follow the chain, you'll find `reserveDoorForExit` → `isSpecificRiderFreeToExit` → `getAiFreeToExit`. It really only needs the status to be set directly before and to be cleared directly after, but I felt it was not as intuitive. Perhaps it is better to do it this way to reduce complexity and the likelihood of unforeseen consequences.

```cpp
rappeller->setStatus(OBJECT_STATUS_IS_USING_ABILITY);
exitInterface->reserveDoorForExit();
rappeller->clearStatus(OBJECT_STATUS_IS_USING_ABILITY);
```

It's unexpected that a door is unavailable when rappelling as indicated by `DEBUG_CRASH(("rappeller is not free to exit... what?"));`.